tests/src/test/java/io/streamnative/pulsar/handlers/kop/TransactionTest.java

```diff
@@ -0,0 +1,277 @@
+            }
+
+            if (txnIndex % 2 != 0) {
+                producer.commitTransaction();
+            } else {
+                producer.abortTransaction();
+            }
+        }
+
+        consumeTxnMessage(topicName, totalTxnCount * messageCountPerTxn, lastMessage, isolation);
+    }
+
+    private void consumeTxnMessage(String topicName,
+                                   int totalMessageCount,
+                                   String lastMessage,
+                                   String isolation) throws InterruptedException {
+        String kafkaServer = "localhost:" + getKafkaBrokerPort();
+
+        Properties consumerProps = new Properties();
+        consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaServer);
+        consumerProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, IntegerDeserializer.class.getName());
+        consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        consumerProps.put(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG, 1000 * 10);
+        consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, "consumer-test");
+        consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        consumerProps.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, isolation);
+
+        @Cleanup
+        KafkaConsumer<Integer, String> consumer = new KafkaConsumer<>(consumerProps);
+        consumer.subscribe(Collections.singleton(topicName));
+
+        log.info("the last message is: {}", lastMessage);
+        AtomicInteger receiveCount = new AtomicInteger(0);
+        while (true) {
+            ConsumerRecords<Integer, String> consumerRecords =
+                    consumer.poll(Duration.of(100, ChronoUnit.MILLIS));
+
+            boolean readFinish = false;
+            for (ConsumerRecord<Integer, String> record : consumerRecords) {
+                log.info("Fetch for receive record offset: {}, key: {}, value: {}",
+                        record.offset(), record.key(), record.value());
+                receiveCount.incrementAndGet();
+                if (lastMessage.equalsIgnoreCase(record.value())) {
+                    log.info("receive the last message");
+                    readFinish = true;
+                }
+            }
+
+            if (readFinish) {
+                log.info("Fetch for read finish.");
+                break;
+            }
+        }
+        log.info("Fetch for receive message finish. isolation: {}, receive count: {}", isolation, receiveCount.get());
+
+        if (isolation.equals("read_committed")) {
+            Assert.assertEquals(receiveCount.get(), totalMessageCount / 2);
+        } else {
+            Assert.assertEquals(receiveCount.get(), totalMessageCount);
+        }
+        log.info("Fetch for finish consume messages. isolation: {}", isolation);
+    }
+
+//    @Test
+    public void test() throws Exception {
```

Comment:
Is the test not ready now?

Reply:
Yes, this test is not ready now. I'll remove this first.